py/torch_tensorrt/dynamo/conversion/impl/embedding.py

```diff
@@ -40,5 +44,125 @@ def embedding(
 
     # Implement embedding lookup with gather layer
     gather_layer = ctx.net.add_gather(embedding_tensor, indices_tensor, axis=0)
-    set_layer_name(gather_layer, target, name + "_gather", source_ir)
+    set_layer_name(gather_layer, target, f"{name}_gather", source_ir)
     return gather_layer.get_output(0)
+
+
+def embedding_bag(
+    ctx: ConversionContext,
+    target: Target,
+    source_ir: Optional[SourceIR],
+    name: str,
+    weight: TRTTensor,
+    indices: TRTTensor,
+    offsets: Optional[Union[torch.Tensor, np.ndarray]],
+    scale_grad_by_freq: bool,
+    mode: int,
+    sparse: bool,
+    per_sample_weights: Optional[TRTTensor],
+    include_last_offset: bool,
+) -> Tuple[TRTTensor, TRTTensor, TRTTensor, TRTTensor]:
+    """
+    This function is for calculating embedding bags.
+    In PyTorch, `offsets` is only used when input is 1D. If input is 2D of shape (B, N), it will be treated as B bags (sequences) each of fixed length N, and this will return B values aggregated in a way depending on the mode. `offsets` is ignored and required to be None in this case.
+    However, according to the schema, `offsets` is required for input with any dimensions. Accordingly, this function flattens N-D input to 1D and then to calculate embedding bags.
+    """
+    if len(indices.shape) != 1:
+        raise RuntimeError("Currently we only support 1D input.")
+        # indices = impl.shuffle.reshape(ctx, target, source_ir, f"{name}_reshape_indices", indices, (-1,))
```

Comment:
I just tried to output `node.args[0].meta`, but it's empty.

Reply:
Since `indices` is just a `placeholder` in the validator, how can we check the shape during compiling?

Reply:
For the test cases, you may need to enable the lowering passes to get the shape prop - try with  `enable_passes=True`

Reply:
Essentially, there is a pass which propagates the tensor shape information through the graph and stores it in the `meta` field

Reply:
Can we assume users always set `enable_passes=True` in this case? If not, `node.args[0].meta` will be empty and capability_validator will pass and then the runtime error will be thrown out because we cannot check the shape during compiling, right?

Reply:
We cannot always assume that `tensor_meta` will exist, so we can include a failsafe in the validator for this (though `tensor_meta` should almost always exist). For instance,
```python
tensor_meta = node.args[1].meta.get("tensor_meta", None)

if tensor_meta is not None:
    return ...
else:
    return False
```
The above will just return False (no conversion), unless that field exists, which in almost all cases it will.

Reply:
Yes, this is what I'm doing for now. But I'm thinking if `enable_passes=False` and the input is 1D, it will not be converted because `tensor_meta is None`, even though our converter is able to handle this scenario.

Reply:
That is true - but it should be a very rare scenario, and it would be better to fall back to Torch in a case we _can_ support than to crash in a case we can't.